### PR TITLE
Improve error message for uninitialized configurable

### LINKED ISSUE #29156: Improve error message for uninitialized configurable
**Description:**
```ballerina
configurable string name;
public function main() {
    panic error("good");
}

```
```
ERROR [main.bal:(1:25,1:25)] configurable variable must be initialized
```

**Describe your problem**
Above error can be misleading. User will think variable has to be initialized.

**Describe your solution**
We should have a message that indicate users can put `?` **or** initialize.

### LINKED ISSUE #29187: Remove unused error for module-level binding pattern support
We cannot code a binding pattern in module level. For example, the following code will give compile error

`[int, int] [a, a] = [54, 54];`

The error is 
`'list' binding pattern not yet supported in module variable declaration`

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$subject

Fixes #29156
Fixes #29187

### CODE REVIEW

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 1]
```diff
@@ -420,9 +420,6 @@ error.type.cast.not.yet.supported.for.type=\
 error.equality.not.yet.supported.for.type=\
   equality not yet supported for type ''{0}''
 
-error.binding.pattern.not.yet.supported.for.type=\
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties:423]: This is not the one relevant for BINDING_PATTERN_NOT_YET_SUPPORTED_IN_MODULE_VAR_DECL right? Seems like both can be removed though.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java [hunk 2]
```diff
@@ -228,8 +228,8 @@
             "error.missing.named.worker.declaration.in.fork.stmt"),
     ERROR_MISSING_KEY_EXPR_IN_MEMBER_ACCESS_EXPR("BCE0514", "error.missing.key.expr.in.member.access.expr"),
     ERROR_MISSING_ERROR_MESSAGE_BINDING_PATTERN("BCE0515", "error.missing.error.message.binding.pattern"),
-    ERROR_CONFIGURABLE_VARIABLE_MUST_BE_INITIALIZED("BCE0516",
-            "error.configurable.variable.must.be.initialized"),
+    ERROR_CONFIGURABLE_VARIABLE_MUST_BE_INITIALIZED_OR_REQUIRED("BCE0516",
+            "error.configurable.variable.must.be.initialized.or.marked.as.required"),
```
Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java:232]: Isn't this usually the same as the code?

File: compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties [hunk 3]
```diff
@@ -218,7 +218,8 @@ error.missing.xml.atomic.name.pattern=missing xml atomic name pattern
 error.missing.tuple.member=missing tuple member
 error.expression.expected.action.found=action invocation as an expression not allowed here
 error.missing.key.expr.in.member.access.expr=missing key expr in member access expr
-error.configurable.variable.must.be.initialized=configurable variable must be initialized
+error.configurable.variable.must.be.initialized.or.marked.as.required=configurable variable must be initialized or \
+                                                                                                    marked as required
```
Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:222]:
```suggestion
                                                                                                    be marked as required
```

File: compiler/ballerina-lang/src/main/resources/compiler.properties (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:423]: Will remove both. thanx

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**